src/main/java/seedu/address/logic/commands/EditGroupCommand.java

```diff
@@ -0,0 +1,120 @@
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.group.Group;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a command that edits the details of an existing group in the address book.
+ */
+public class EditGroupCommand extends Command {
+
+    /** The command word to trigger this command. */
+    public static final String COMMAND_WORD = "edit-group";
+
+    /** Usage message for the command. */
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the group identified "
+            + "by the index number used in the displayed group list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "CS2103T T12-1 ";
+
+    /** Success message for editing a group. */
+    public static final String MESSAGE_EDIT_GROUP_SUCCESS = "Edited Group: %1$s";
+
+    /** Error message if a duplicate group exists in the address book. */
+    public static final String MESSAGE_DUPLICATE_GROUP = "This group already exists in the address book.";
+
+    /** Index of the group to be edited. */
+    private final Index index;
+
+    /** New name for the group. */
+    private final String newGroupName;
+
+    /**
+     * Creates an EditGroupCommand to update a group's name.
+     *
+     * @param index The index of the group to be edited.
+     * @param newGroupName The new name for the group.
+     */
+    public EditGroupCommand(Index index, String newGroupName) {
+        requireAllNonNull(index, newGroupName);
+
+        this.index = index;
+        this.newGroupName = newGroupName;
+    }
+
+    /**
+     * Executes the command to edit a group.
+     *
+     * @param model The model in which the command should be executed.
+     * @return A CommandResult indicating the outcome of the command execution.
+     * @throws CommandException If the index is invalid or if the new group name already exists.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Group> lastShownList = model.getFilteredGroupList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException("Invalid Group");
+        }
+
+        Group groupToEdit = lastShownList.get(index.getZeroBased());
+        Group editedGroup = createEditedGroup(groupToEdit, newGroupName);
+
+        if (model.hasGroup(editedGroup)) {
+            throw new CommandException(MESSAGE_DUPLICATE_GROUP);
+        }
```

Comment:
It's a bit unnecessary to check for duplicates here when UniqueGroupList already does so for us. But I suppose that this mirrors the logic for the EditPersonCommand, so I think it's fine to keep it the same.